### DISCUSSION TITLE
Email Attachment Content IDs

### DISCUSSION THREAD

**Discussion by agentabstract** [microsoftgraph/msgraph-sdk-go#767] [General]
When I list the attachments of an email I get the interface type models.AttachmentCollectionResponseable,  I list the values of that and it returns []models.Attachmentable.  The interface models.Attachmentable allows me to see that an attachment is inline,  but has no method to get the content id of the file.  

How do you get the content id of the file using the go sdk?

According to the docs Content ID should be in the HTTP Response.

`https://learn.microsoft.com/en-us/graph/api/message-list-attachments?view=graph-rest-1.0`

**Comment by baywet** [comment]
Hi @agentabstract 
Thanks for using the SDK and for reaching out.
Have you tried a type assertion to file attachment? 
https://learn.microsoft.com/en-us/graph/api/resources/fileattachment?view=graph-rest-1.0

**Reply by agentabstract** [reply]
I have not.  

What I ended up doing was the  users.ItemPhotoValueContentRequestBuilderGetRequestConfiguration trick, with the URL "https://graph.microsoft.com/v1.0/users/[EMAIL_ACCOUNT]/messages/[MESSAGEID]/attachments".  

When I ask for the attachments list this way I get the content id in the JSON and unmarshal it. It also sends me the bytes of the attachment in the JSON, where in the past I would iterate over the list downloading each attachment also having to use the users.ItemPhotoValueContentRequestBuilderGetRequestConfiguration trick.  

```rawUrl := fmt.Sprintf("https://graph.microsoft.com/v1.0/users/%v/messages/%v/attachments", m.MailBoxEmail, m.MSMessageID)
j := &users.ItemPhotoValueContentRequestBuilderGetRequestConfiguration{}
j.Headers = &abstractions.RequestHeaders{}
j.Headers.Add("Prefer", "IdType=\"ImmutableId\"")
builder := users.NewItemPhotoValueContentRequestBuilder(rawUrl, appClient.RequestAdapter)
attachmentValue, err := builder.Get(innerCTX, j)
if err != nil {
     return nil, err
}
type msAttachments struct {
  Value []struct {
	  ContentType  string `json:"contentType"`
	  ID           string `json:"id"`
	  Name         string `json:"name"`
	  Size         int32  `json:"size"`
	  IsInline     bool   `json:"isInline"`
	  ContentBytes []byte `json:"contentBytes"`
	  ContentID    string `json:"contentId"`
  } `json:"value"`
}
var list msAttachments
err = json.Unmarshal(attachmentValue, &list)
if err != nil {
     return nil, err
}
```
